core/arch/arm/kernel/spmc_sp_handler.c

```diff
@@ -1257,7 +1251,7 @@ static void handle_console_log(struct thread_smc_args *args)
 		goto out;
 
 	for (n = 0; n < char_count; n += reg_size)
-		memcpy(buffer + n, reg_list[n / reg_size],
+		memcpy(buffer + n, &args->a[n / reg_size],
```

Comment:
This is not correct, the 'Character list' parameter starts from W2/X2. Now it tries to print the function id and character count fields too. Adding 2 to the index solves the issue.

Reply:
Good catch, I'll update.

Reply:
The updated version fixed the issue, thanks.